### DISCUSSION TITLE
gatsby-plugin-s3 maintenance and contribution

### DISCUSSION THREAD

**Discussion by davidrhoderick** [gatsbyjs/gatsby#37695] [Help]
I've been able to follow the [documentation here](https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/deploying-to-s3-cloudfront/) to use the `gatsby-plugin-s3` to deploy to S3.  It's pretty awesome because there is so little configuration required nor any need to traverse through folders to upload files to S3.  It got us to PoC pretty quickly on our project.

However, if you look at [this issue I've submitted](https://github.com/jariz/gatsby-plugin-s3/issues/422), it documents how this plugin has fallen out of maintenance.  It's not possible to run it in its current state following the contribution guidelines.

If we are going to be using this in production, I think it should be maintained or at least in a maintainable state.  I'd be willing to take it over but I just thought I'd reach out to the Gatsby team because it is referenced in the official documentation as the accepted way to upload to S3 + CloudFront.

1. Is there a more modern & supported approach to S3 + CloudFront uploading that we should adopt?
2. Is there anyone who might be able to bring this project into a more maintainable state?  For example, can someone more quickly update the contribution guidelines or the project so they work for open source development?
3. If neither of those, should we choose to fork this repo and figure it out, should the documentation point to our maintained version?

Thanks for any direction you can point us.

**Comment by LekoArts** [comment]
If you're interested in contributing, you could look into bringing the plugin over to https://github.com/gatsby-uc/plugins and maintain it there

**Reply by davidrhoderick** [reply]
I'm asking them now about how to proceed if we don't have official maintainer initiation.  I guess the fundamental issue with code that goes unmaintained is it becomes difficult to get ahold of the maintainer in order to contribute.
